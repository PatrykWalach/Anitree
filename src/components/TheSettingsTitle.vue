<template>
  <v-list-item @click.stop>
    <component
      v-model="dialog"
      :is="$vuetify.breakpoint.xsOnly ? 'v-bottom-sheet' : 'v-dialog'"
      :max-width="$vuetify.breakpoint.xsOnly ? undefined : '320px'"
    >
      <template v-slot:activator="{ on, attrs }">
        <v-list-item-content v-on="on" v-bind="attrs">
          <v-list-item-title class="text-capitalize"
            >Title language - {{ preferedTitle }}</v-list-item-title
          >
          <v-list-item-subtitle
            >Select your prefered language for all the
            titles</v-list-item-subtitle
          >
        </v-list-item-content>
      </template>

      <v-card>
        <v-card-title>
          Select prefered language
        </v-card-title>
        <v-divider></v-divider>
        <v-list>
          <v-list-item-group @change="changePreferedTitle">
            <v-list-item v-for="title in titles" :value="title" :key="title">
              <v-list-item-content>
                <v-list-item-title>
                  {{ title }}
                </v-list-item-title>
              </v-list-item-content>
            </v-list-item>
          </v-list-item-group>
        </v-list>
      </v-card>
    </component>
  </v-list-item>
</template>

<script lang="ts">
import { createComponent, ref } from '@vue/composition-api'
import useTitle from '@/store/title'
import { VDialog, VBottomSheet } from 'vuetify/lib'

export default createComponent({
  components: {
    VDialog,
    VBottomSheet
  },
  setup() {
    const dialog = ref(false)
    const { preferedTitle, titles } = useTitle()

    const changePreferedTitle = (title: 'native' | 'romaji' | 'english') => {
      dialog.value = false
      preferedTitle.value = title
    }

    return {
      preferedTitle,
      titles,
      dialog,
      changePreferedTitle
    }
  }
})
</script>
