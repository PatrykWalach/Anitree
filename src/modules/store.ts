import * as modules from '.'
import { Store, plugin } from 'vuex-composition-api'

export const store =
  process.env.NODE_ENV === 'production'
    ? undefined
    : new Store({
        plugins: [plugin(Object.values(modules))],
      })
