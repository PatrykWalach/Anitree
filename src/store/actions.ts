export { titleActions as title } from './modules/title'

export { editActions as edit } from './modules/edit'
export { filterActions as filter } from './modules/filter'
export { shareActions as share } from './modules/share'
export { settingsActions as settings } from './modules/settings'
export { timelineActions as timeline } from './modules/timeline'
export { changesActions as changes } from './modules/changes'
