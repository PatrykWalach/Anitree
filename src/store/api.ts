import gql from 'graphql-tag'
import { FetchVariables } from './models'

import apollo from '../apollo'
import { AMedia } from '@/types'

export const query = gql`
  query($search: String, $idIn: [Int], $page: Int = 0) {
    Page(page: $page, perPage: 50) {
      media(search: $search, id_in: $idIn) {
        id
        bannerImage
        coverImage {
          extraLarge
          large
          medium
          color
        }
        format
        status
        tags {
          name
          id
          rank
        }
        siteUrl
        type
        studios(isMain: true) {
          nodes {
            name
            id
          }
        }
        startDate {
          year
          month
          day
        }
        endDate {
          year
          month
          day
        }
        title {
          romaji
          english
          native
        }
        description
        mediaListEntry {
          status
        }
        relations {
          edges {
            node {
              id
              type
            }
            id
            relationType
          }
        }
      }
    }
  }
`

export function fetchMediaApollo(variables: FetchVariables): Promise<AMedia[]> {
  return apollo
    .query<{ Page: { media: AMedia[] } }>({
      query,
      variables
    })
    .then(({ data }) => data.Page.media)
}
