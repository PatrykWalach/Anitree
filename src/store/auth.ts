import VueFunctionApi, { ref, Ref, computed } from 'vue-function-api'
import Vue from 'vue'
Vue.use(VueFunctionApi)

import { Token } from '@/types'
const stored = localStorage.getItem('AUTH')

const auth: Ref<Token | null> = ref((stored && JSON.parse(stored)) || null)

const token = computed(() => {
  if (
    auth.value
    // && Date.now() < parseInt(auth.value.expires_in)
  ) {
    return auth.value.access_token
  }
  return null
})

const CHANGE_TOKEN = async (payload: Token | null) => {
  if (payload) {
    const newAuth: Token = {
      ...payload,
      /* eslint-disable-next-line @typescript-eslint/camelcase */
      expires_in: (Date.now() + parseInt(payload.expires_in)).toString()
    }
    localStorage.setItem('AUTH', JSON.stringify(newAuth))
    auth.value = newAuth
  } else {
    auth.value = payload
  }
}

export default function useAuth() {
  return { auth, CHANGE_TOKEN, token, stored }
}
