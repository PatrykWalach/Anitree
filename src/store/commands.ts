import CompositionApi, { Ref, ref, watch } from '@vue/composition-api'
import {
  DeleteCommand,
  DeleteCommandConstructor,
} from './commands/DeleteCommand'
import { DeleteVariables, SaveVariables } from '@/graphql/schema/listEntry'
import { SaveCommand, SaveCommandConstructor } from './commands/SaveCommand'
import { Form } from '@/types'
import { FuzzyDate } from '@/graphql/schema/media'
import { MediaList } from '@/graphql/schema/mediaListCollection'
import Vue from 'vue'
import { useSettings } from './settings'

Vue.use(CompositionApi)

export interface Command {
  saveState?(): void | Promise<void>
  undo?(): void | Promise<void>
  execute(): void | Promise<void>
}

export interface ListCommand extends Command {
  backup: Form | null
  variables: SaveVariables | (DeleteVariables & Pick<SaveVariables, 'mediaId'>)
  done: boolean
  _class: 'SaveCommand' | 'DeleteCommand'
}

export const isDeleteCommand = (
  e: (SaveCommandConstructor | DeleteCommandConstructor) &
    Pick<ListCommand, '_class'>,
): e is DeleteCommandConstructor & Pick<DeleteCommand, '_class'> =>
  e._class === 'DeleteCommand'

export const isSaveCommand = (
  e: (SaveCommandConstructor | DeleteCommandConstructor) &
    Pick<ListCommand, '_class'>,
): e is SaveCommandConstructor & Pick<SaveCommand, '_class'> =>
  e._class === 'SaveCommand'

const stored = localStorage.getItem('CHANGES')

const history: Ref<ListCommand[]> = ref(
  (stored &&
    JSON.parse(stored, (key, value) => {
      if (value instanceof Object) {
        if (isSaveCommand(value)) {
          return new SaveCommand(value)
        }
        if (isDeleteCommand(value)) {
          return new DeleteCommand(value)
        }
      }
      return value
    })) ||
    [],
)

watch(() => {
  if (useSettings().cacheChanges.value) {
    localStorage.setItem('CHANGES', JSON.stringify(history.value))
  }
})
;(async () => {
  for (const command of history.value.filter(({ done }) => !done)) {
    await command.execute()
  }
})()

const dispatch = async (command: ListCommand) => {
  history.value.push(command)
  await command.execute()
  return command
}

export const useCommands = () => {
  return {
    dispatch,
    history,
  }
}

export const mediaListToForm = (
  mediaListEntry: MediaList | null,
  advancedScoring: string[],
): Form => {
  if (mediaListEntry) {
    const advancedScores = advancedScoring
      .map(key => mediaListEntry.advancedScores[key])
      .map(value => value || 0)

    delete mediaListEntry.__typename
    delete mediaListEntry.startedAt.__typename
    delete mediaListEntry.completedAt.__typename
    delete mediaListEntry.private
    delete mediaListEntry.id
    delete mediaListEntry.mediaId

    return {
      ...mediaListEntry,
      advancedScores,
    }
  }

  const advancedScores = advancedScoring.map(() => 0)
  return {
    advancedScores,
    completedAt: { day: null, month: null, year: null },
    notes: '',
    progress: 0,
    progressVolumes: 0,
    repeat: 0,
    score: 0,
    startedAt: { day: null, month: null, year: null },
    status: null,
  }
}

export const isDate = (e: any): e is FuzzyDate => e.__typename === 'FuzzyDate'
