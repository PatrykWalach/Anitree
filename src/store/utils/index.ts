export * from './useCommands'
export * from './useTitle'
export * from './useEdit'
