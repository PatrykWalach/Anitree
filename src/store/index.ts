import Vue from 'vue'
import Vuex from 'vuex'

// import { getModule } from 'vuex-module-decorators'
Vue.use(Vuex)

import { ModuleMedia } from './modules/media'
import { ModuleTheme } from './modules/theme'
import { ModuleTitle } from './modules/title'
import { ModuleAuth } from './modules/auth'

export interface RootState {
  media: ModuleMedia
  theme: ModuleTheme
  title: ModuleTitle
  auth: ModuleAuth
}

export default new Vuex.Store<RootState>({})
