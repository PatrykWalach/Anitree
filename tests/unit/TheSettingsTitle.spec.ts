import TheSettingsTitle from '@/components/TheSettingsTitle.vue'
import { matchSnapshot } from './utils'
import { useMockedStore } from './mocks/store'

describe('TheSettingsTitle', () => {
  matchSnapshot(TheSettingsTitle, useMockedStore())
})
