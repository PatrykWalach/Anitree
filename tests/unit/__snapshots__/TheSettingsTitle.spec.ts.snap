// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSettingsTitle matches snapshot 1`] = `
<v-list-item-stub
  activeclass=""
  tag="div"
>
  <v-dialog
    max-width="320px"
  >
     
    <v-card-stub
      loaderheight="4"
      tag="div"
    >
      <v-card-title-stub>
        
        Title language
      
      </v-card-title-stub>
       
      <!---->
       
      <v-list-stub
        flat="true"
        tag="div"
        tile="true"
      >
        <v-list-item-group-stub
          activeclass="v-item--active"
          value="romaji"
        >
          <v-list-item-stub
            activeclass=""
            tag="div"
            value="romaji"
          />
          <v-list-item-stub
            activeclass=""
            tag="div"
            value="english"
          />
          <v-list-item-stub
            activeclass=""
            tag="div"
            value="native"
          />
        </v-list-item-group-stub>
      </v-list-stub>
       
      <v-card-actions-stub>
        <v-spacer-stub />
         
        <v-btn-stub
          activeclass=""
          tag="button"
          text="true"
          type="button"
        >
          cancel
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog>
</v-list-item-stub>
`;
